sched/scheduler/stateful_scheduler.go

```diff
@@ -870,23 +851,9 @@ func (s *statefulScheduler) scheduleTasks() {
 							"jobType":     jobType,
 							"tag":         tag,
 						}).Info("Task *node* lost, cleaning up.")
-				} else if nodeSt.runningJob != jobID || nodeSt.runningTask != taskID {
```

Comment:
I'm not entirely sure if this is just for P3. Do you think the if-statement above is sufficient to handle the node lost case?

Reply:
On second thought, let's trust the comment at face value.

That said, I think what's bothering me is the handling of jobState.numTimesTried if a node hiccups and comes back. We may need to keep the 'preempted' arg and set it to true if the nodeId is the same but the actual node object has been replaced.